pkg/controllers/disruption/helpers.go

```diff
@@ -145,15 +145,63 @@ func GetPodEvictionCost(ctx context.Context, p *v1.Pod) float64 {
 	return clamp(-10.0, cost, 10.0)
 }
 
-func filterByPrice(options []*cloudprovider.InstanceType, reqs scheduling.Requirements, price float64) []*cloudprovider.InstanceType {
+// filterByPriceWithMinValues returns the instanceTypes that are lower priced than the current candidate and iterates over the cumulative minimum requirement of the InstanceTypeOptions to see if it meets the minValues of requirements.
+// The minValues requirement is checked again after filterByPrice as it may result in more constrained InstanceTypeOptions for a NodeClaim
+func filterByPriceWithMinValues(options []*cloudprovider.InstanceType, reqs scheduling.Requirements, price float64) ([]*cloudprovider.InstanceType, string) {
 	var result []*cloudprovider.InstanceType
+	// cumulativeMinRequirementsFromInstanceTypes is a map for the requirement key with the cumulative values that has minValues supported across InstanceTypeOptions
```

Comment:
You've given a super thorough explanation of what this variable is tracking here, which I think is great! I wonder if you should do the same thing in the `provisioning/scheduling/nodeclaim.go` where we are basically performing the same logic

Reply:
Initially I thought of adding this at both the places but given the verbosity, I thought a single place was sufficient. Added it at the other place as well.